python/common/org/python/types/Float.java

```diff
@@ -305,37 +305,48 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
         } else if (other instanceof org.python.types.Complex) {
             return ((org.python.types.Complex) other).__mul__(this);
-        } else if (other instanceof org.python.types.Dict) {
-            throw new org.python.exceptions.TypeError("unsupported operand type(s) for *: 'float' and '" + other.typeName() + "'");
-        } else if (other instanceof org.python.types.NoneType) {
-            throw new org.python.exceptions.TypeError("unsupported operand type(s) for *: 'float' and '" + other.typeName() + "'");
-        } else if (other instanceof org.python.types.Set) {
-            throw new org.python.exceptions.TypeError("unsupported operand type(s) for *: 'float' and '" + other.typeName() + "'");
-        } else if (other instanceof org.python.types.FrozenSet) {
-            throw new org.python.exceptions.TypeError("unsupported operand type(s) for *: 'float' and '" + other.typeName() + "'");
-        } else if (other instanceof org.python.types.List) {
-            throw new org.python.exceptions.TypeError("can't multiply sequence by non-int of type 'float'");
-        } else if (other instanceof org.python.types.Tuple) {
-            throw new org.python.exceptions.TypeError("can't multiply sequence by non-int of type 'float'");
-        } else if (other instanceof org.python.types.Slice) {
-            throw new org.python.exceptions.TypeError("unsupported operand type(s) for *: 'float' and 'slice'");
-        } else if (other instanceof org.python.types.ByteArray){
-            throw new org.python.exceptions.TypeError("can't multiply sequence by non-int of type 'float'");
+        } else {
+            super.__mul__(other);
         }
 
         throw new org.python.exceptions.NotImplementedError("float.__mul__() has not been implemented.");
     }
 
+
+    @org.python.Method(
+            __doc__ = "Return self*value.",
+            args = {"other"}
+    )
+    public org.python.Object __imul__(org.python.Object other) {
+
+        if (org.python.types.Object.isSequence(other)) {
+            throw new org.python.exceptions.TypeError("can't multiply sequence by non-int of type '" + "float" + "'");
+        } else if (other instanceof org.python.types.Int) {
+            return new org.python.types.Float(this.value * ((org.python.types.Int) other).value);
+        } else if (other instanceof org.python.types.Float) {
+            return new org.python.types.Float(((double) this.value) * ((org.python.types.Float) other).value);
+        } else if (other instanceof org.python.types.Bool) {
+            return new org.python.types.Float(this.value * (((org.python.types.Bool) other).__int__().value));
+        } else if (other instanceof org.python.types.Complex) {
+            return ((org.python.types.Complex) other).__imul__(this);
+        } else {
+            super.__imul__(other);
+        }
+        throw new org.python.exceptions.NotImplementedError("float.__mul__() has not been implemented.");
```

Comment:
This doesn't look right -- if we're adding an implementation, why are we throwing NotImplementedError here?
Let's add a test for a different type (say, passing an empty string `''`) and then catch and print the error -- this probably needs to throw TypeError.

Reply:
@eliasdorneles  Oh yes, I made a terrible mistake there. Removing exception throwing will do right ?

Reply:
To be fair, it looks like the mistake already existed in `__mul__` implementation.

> Removing exception throwing will do right ?

I suggest adding a test for the case and let the test tell you what to do (run it, see the failure, figure out what the output should be). :)

Reply:
Hi, I actually added test case and it worked(```TypeError``` thing). Please have a look at [this](https://github.com/pybee/voc/pull/751/files#diff-0eec62afc15f3b0631bd40cb3d44b3af). I think I didn't understand the problem in the first place, could you please give me some hints ? Thank you.